core/domain/services/assets/ReactAssetManager.php

```diff
@@ -35,6 +35,17 @@ class ReactAssetManager extends AssetManager
      */
     public function addAssets()
     {
+        // TODO get these values from router
+        $is_ee_domain = true;
+        $is_blocks_domain = false;
+
+        if (!$is_ee_domain || $is_blocks_domain) {
```

Comment:
This will almost certainly not work because it's nearly impossible to detect a "blocks" domain since users could put them nearly anywhere. We will most likely have to load blocks code for every frontend request.
 
 Other domains can be more targeted.

Reply:
Alright, then may be make it just `if (!$is_ee_domain) return`

Reply:
well it won't matter now, because we are not implementing any frontend domains, but in the future it may be possible to have an overlap between blocks and other domains, although there are steps we could potentially take to avoid that. My point is that it may be tricky to try and implement a separate version of React than what core is using. Is there any critical reason we need to upgrade our version of React?

Reply:
> Is there any critical reason we need to upgrade our version of React

Yes, sometimes we rely on the latest React API and it won't be possible for us to be sure that the API will be available for a specific version of WP.